### ISSUE TITLE
[Enhancement] Return total row count in GET /trails endpoint

### ISSUE THREAD

**Issue by vjaitly** [nearform/trail#57]
The GET /trails endpoint supports paginated requests, but for it to properly work for a UI that we are building on top of the API , the total row count is also required. 
In our use case, we could have huge number of audit logs, hence paginated requests is the only option.

**Comment by simoneb** [comment]
@vjaitly would you mind sending some PRs for this and the other issues you have created? We would appreciate it very much and we can support you if you need any input.

**Comment by vjaitly** [comment]
@simoneb I can give it a try. This one is a little tricky though. The current implementation is based on limit/offset and not a cursor, hence it seems the query would have to be called twice to get the total count.
Other simpler option would be to have a separate API just to get the count - this gives the flexibility to the client to use the total count or not and not penalising them in case they do not use the pagination functionality.
Any inputs on how you would envisage the implementation would be helpful.

Also since you have the hapi and GraphQL implementation too, i am assuming you would want the APIs in all the variation to remain same.

**Comment by simoneb** [comment]
The queries live in core so the change could be there, even though the other packages would also need to be updated. The idea would be to send two queries, one is the existing one, and another for getting the whole row count.

**Comment by vjaitly** [comment]
@simoneb The response schema for /trails GET API could change to something like this:
```
S.object()
          .prop('count', S.number()
            .description('The search result count')
          )
          .prop('data', S.array()
            .description('The search results.')
            .items(trailSchema.response)
          )
```
However this would mean a breaking change for the APIs in the other packages as well. The query change would be in the core, but that would also break the other packages as the response in the `search()` method of core will change too. Also all the tests related to `search()` will need an update too.

**Comment by vjaitly** [comment]
Have created a pull request with the assumption that the above change is acceptable. #108

**Comment by simoneb** [comment]
Closed via #108
